src/WriterRegistrar.h

```diff
@@ -0,0 +1,76 @@
+// SPDX-License-Identifier: BSD-2-Clause
+//
+// This code has been produced by the European Spallation Source
+// and its partner institutes under the BSD 2 Clause License.
+//
+// See LICENSE.md at the top level for license information.
+//
+// Screaming Udder!                              https://esss.se
+
+/// \brief Keeps track of the registered FlatbufferReader instances.
+///
+/// Writer modules register themselves via instantiation of a
+/// `WriterModule::Registry::Registrar`.
+
+#pragma once
+#include "WriterModuleBase.h"
+#include <map>
+#include <string>
+
+namespace WriterModule {
+namespace Registry {
+using ModuleFactory = std::function<std::unique_ptr<WriterModule::Base>()>;
+using IdNameHash = size_t;
+
+IdNameHash getWriterModuleHash(std::string const &FlatbufferID,
+                               std::string const &ModuleName);
+
+/// \brief Get all registered modules.
+///
+/// \return A reference to the map of registered modules.
+std::map<std::string, std::string> getFactoryIdsAndNames();
+
+/// \brief Registers a new writer module. Called by `Registrar`.
+///
+/// \param key
+/// \param value
+void addWriterModule(std::string const &FlatbufferID,
+                     std::string const &ModuleName, ModuleFactory Value);
+
+/// \brief Get `ModuleFactory for a given flatbuffer id.
+///
+/// \return Matching `ModuleFactory`.
+ModuleFactory const &find(std::string const &FlatbufferID);
+
+/// \brief Get module factory for a given flatbuffer id and/or module name.
+/// \param[in] FlatbufferID A four character flatbuffer id. Will be ignored if
```

Comment:
Can we lose the [in]s please? It is unnecessary noise in most cases.
A new line after the \brief makes my rapidly failing eyes happy.

Generally, I think we should aim to keep the doxygen as simple as possible then there is less chance of it being out of date.

Obviously it is a different story if we are supplying a public API...